WordPress/gutenberg
# [Block Library - Social Links]: Use the new `flex` layout

## Description
Related to: https://github.com/WordPress/gutenberg/pull/33359
Fixes: https://github.com/WordPress/gutenberg/issues/34003

This PR explores the new `flex layout` to be used by `Social Links` block. The goal is to use `flex` layout in a declarative way and not rely on 'hackish` css custom solutions for blocks that a `flex` layout is a good fit.


## Notes
1. I've [found a bug](https://github.com/WordPress/gutenberg/issues/34003) in `Social Icons` block regarding `itemsJustification` attribute which was introduced in https://github.com/WordPress/gutenberg/pull/28980 and wasn't declared in block.json. With this PR the issue will be resolved as we remove it's usage.

2. I observed that also on `trunk` if you select the block, the alignment is different than the one applied at the end (it uses `flex-start`. This PR doesn't have this problem.
3. The `justify content` control will need polishing in regards to `labels` etc..
4. Tests will be updated and/or created..
5. A small visual difference is that `gap` is applied now due to this previous change: https://github.com/WordPress/gutenberg/pull/34493/files#diff-324697e41855298e2f2c74b078f174e0cbc9075cef736ce9c1e2c169bf64652eR70. 


## Testing instructions
1. Insert `Social Links` block and in Inspector controls play with the justify content control.
2. Check that previously created `Social Links` blocks are displayed as before in both editor and front-end.

## Tasks
 - [ ] Check native file
